### PR TITLE
Type Easings and Colors

### PR DESCRIPTION
**Pull request by jakub-gonet**
## Description

Add some types to Easing, Colors, Bezier.

### CODE REVIEW

File: src/reanimated2/Colors.ts [hunk 1]
```diff
@@ -7,18 +7,29 @@
...
+  r: number;
+  g: number;
+  b: number;
+};
+
+type HSVObj = {
+  h: number;
+  s: number;
+  v: number;
+};
```
Comment (mrousavy) [inline src/reanimated2/Colors.ts:26]: Is there a specific reason for making those things `type`? `interface`s have a few advantages over types, IIRC that's for performance and decl merging.

so,

```ts
interface RGB {
  r: number;
  g: number;
  b: number;
}
```


is always better unless you have a specific reason

Comment (jakub-gonet) [reply src/reanimated2/Colors.ts:26]: Mosty muscle memory after I stumbled upon this issue: https://github.com/microsoft/TypeScript/issues/15300

This code can easily be swapped for interfaces, thanks for noticing.

Comment (jakub-gonet) [reply src/reanimated2/Colors.ts]: Fixed in c2dfc60.

File: src/reanimated2/Colors.ts [hunk 2]
```diff
@@ -7,18 +7,29 @@
...
+  s: number;
+  v: number;
+}
+
 // var INTEGER = '[-+]?\\d+';
 const NUMBER = '[-+]?\\d*\\.?\\d+';
 const PERCENTAGE = NUMBER + '%';
 
-function call(...args) {
+function call(...args: unknown[]) {
```
Comment (piaskowyk) [inline src/reanimated2/Colors.ts:32]: It seems the return type is missing. `: string`

File: src/reanimated2/Colors.ts [hunk 3]
```diff
@@ -493,12 +528,16 @@ function RGBtoHSV(r, g, b) {
  * 0 <= h, s, v <= 1
  * returns 0 <= r, g, b <= 255
  */
-function HSVtoRGB(h, s, v) {
+function HSVtoRGB(hsv: HSV): RBG;
+function HSVtoRGB(h: number, s: number, v: number): RBG;
+function HSVtoRGB(h: any, s?: any, v?: any) {
```
Comment (piaskowyk) [inline src/reanimated2/Colors.ts:533]: missing type `: RGB`

File: src/reanimated2/Colors.ts [hunk 4]
```diff
@@ -559,15 +604,15 @@ export function processColorInitially(color) {
...
+export function isColor(value: unknown): boolean {
   'worklet';
   if (typeof value !== 'string') {
     return false;
   }
   return processColorInitially(value) != null;
 }
 
-export function processColor(color) {
+export function processColor(color: unknown): number | null | undefined {
```
Comment (piaskowyk) [inline src/reanimated2/Colors.ts:615]: Why color is as `:unknown` type?

File: src/reanimated2/Colors.ts (hunk 2 above)
Comment (jakub-gonet) [reply src/reanimated2/Colors.ts:32]: Thanks, I must've missed that.

File: src/reanimated2/Colors.ts (hunk 4 above)
Comment (jakub-gonet) [reply src/reanimated2/Colors.ts:615]: It uses `processColorInitially` which uses `normalizeColor`, both of them can accept arbitrary arguments (they're returning nulls when they cannot process the arguments). I think typings should reflect that.

Right now, we type this function as:
`export function processColor(color: number | string): number;`
so we can also simplify those methods by using `color: number | string`, casting it internally to unknown (to support JS users too), and e.g. throw when input is invalid. What do you think?

File: src/reanimated2/Colors.ts (hunk 2 above)
Comment (jakub-gonet) [reply src/reanimated2/Colors.ts]: Fixed in e8d1243a.

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**